lib/archive/unpack.go

```diff
@@ -73,64 +67,54 @@ func Unpack(op trace.Operation, tarStream io.Reader, filter *FilterSpec, unpackP
 			// This indicates the end of the archive
 			break
 		}
+
 		if err != nil {
-			// it is likely in this case that we were not given a legitimate tar stream
-			op.Debugf("Received error (%s) when attempting a tar read operation on target stream", err)
+			op.Errorf("Error reading tar header: %s", err)
 			return err
 		}
 
+		op.Debugf("processing tar header: %s", header.Name)
+
 		// skip excluded elements unless explicitly included
-		if excluded(header.Name, filter) {
+		if filter.Excludes(op, header.Name) {
 			continue
 		}
 
 		// fix up path
-		strippedTargetPath := strings.TrimPrefix(header.Name, strip)
-		writePath := filepath.Join(finalTargetPath, strippedTargetPath)
+		stripped := strings.TrimPrefix(header.Name, filter.StripPath)
+		rebased := filepath.Join(filter.RebasePath, stripped)
+		absPath := filepath.Join(root, rebased)
 
 		switch header.Typeflag {
 		case tar.TypeDir:
-			err = os.MkdirAll(writePath, header.FileInfo().Mode())
+			err = os.MkdirAll(absPath, header.FileInfo().Mode())
 			if err != nil {
 				return err
 			}
 			continue
 		case tar.TypeSymlink:
 
-			// NOTE: symbolic links cannot span mounts.
-			// This can cause us to create a regular file instead of
-			// a symlink. this behavior should be evaluated for a proper
-			// response, for now a regular file is made since that is the
-			// target of a sym link operation
-			err := os.Symlink(header.Linkname, writePath)
-			if err == nil {
-				continue
+			err := os.Symlink(header.Linkname, absPath)
+			if err != nil {
+				op.Errorf("Failed to create symlink %s->%s: %s", absPath, header.Linkname, err)
+				return err
```

Comment:
Should we be bailing out of the entire operation due to the inability to make one symlink? Or should we just log this and then write it as a regular file empty file?